web-api/src/storage/postgres.rs

```diff
@@ -145,6 +140,56 @@ impl Database {
                 .await?;
         }
 
+        let mut snippets =
+            documents
+                .iter()
+                .flat_map(|document| {
+                    document.snippets.iter().enumerate().map(
+                        |(snippet_idx, (snippet, embedding))| {
+                            (
+                                &document.id,
+                                #[allow(clippy::cast_possible_truncation)]
+                                SqlBitCastU32::from(snippet_idx as u32),
+                                snippet,
+                                embedding,
+                            )
+                        },
+                    )
+                })
+                .peekable();
+
+        let mut builder = QueryBuilder::new(
+            "INSERT INTO snippet (
+                        document_id,
+                        snippet_idx,
+                        snippet,
+                        embedding
+                    ) ",
+        );
+
+        while snippets.peek().is_some() {
```

Comment:
❓/:art: couldn't we also use the new chunk iterator approach for this loop?

Reply:
What I added is a chunk iteration helper over things which are tuples, not a generic one.  But with how I changed it later it's not much different from a generic one, so I guess yes :thinking: